diff --git a/node_modules/@remix-run/react/dist/data.js b/node_modules/@remix-run/react/dist/data.js
index 98e87d4..4658d6b 100644
--- a/node_modules/@remix-run/react/dist/data.js
+++ b/node_modules/@remix-run/react/dist/data.js
@@ -54,6 +54,7 @@ async function fetchData(request, routeId, retry = 0) {
   let url = new URL(request.url);
   url.searchParams.set("_data", routeId);
   let init = {
+    headers: request.headers,
     signal: request.signal
   };
   if (request.method !== "GET") {
diff --git a/node_modules/@remix-run/react/dist/esm/data.js b/node_modules/@remix-run/react/dist/esm/data.js
index c6728c9..6e32d62 100644
--- a/node_modules/@remix-run/react/dist/esm/data.js
+++ b/node_modules/@remix-run/react/dist/esm/data.js
@@ -50,6 +50,7 @@ async function fetchData(request, routeId, retry = 0) {
   let url = new URL(request.url);
   url.searchParams.set("_data", routeId);
   let init = {
+    headers: request.headers,
     signal: request.signal
   };
   if (request.method !== "GET") {
