diff --git a/node_modules/@remix-run/react/dist/data.js b/node_modules/@remix-run/react/dist/data.js
index d081f36..bee5e3a 100644
--- a/node_modules/@remix-run/react/dist/data.js
+++ b/node_modules/@remix-run/react/dist/data.js
@@ -50,6 +50,7 @@ async function fetchData(request, routeId, retry = 0) {
   let url = new URL(request.url);
   url.searchParams.set("_data", routeId);
   let init = {
+    headers: request.headers,
     signal: request.signal
   };
   if (request.method !== "GET") {
diff --git a/node_modules/@remix-run/react/dist/esm/data.js b/node_modules/@remix-run/react/dist/esm/data.js
index 9b8f047..1a562dd 100644
--- a/node_modules/@remix-run/react/dist/esm/data.js
+++ b/node_modules/@remix-run/react/dist/esm/data.js
@@ -46,6 +46,7 @@ async function fetchData(request, routeId, retry = 0) {
   let url = new URL(request.url);
   url.searchParams.set("_data", routeId);
   let init = {
+    headers: request.headers,
     signal: request.signal
   };
   if (request.method !== "GET") {
