diff --git a/node_modules/@remix-run/router/dist/router.js b/node_modules/@remix-run/router/dist/router.js
index 3d64a32..a1cb1a0 100644
--- a/node_modules/@remix-run/router/dist/router.js
+++ b/node_modules/@remix-run/router/dist/router.js
@@ -3290,6 +3290,7 @@ function normalizeNavigateOptions(normalizeFormMethod, isFetcher, path, opts) {
           formAction,
           formEncType: opts.formEncType,
           formData: undefined,
+          headers: opts.headers,
           json: undefined,
           text
         }
@@ -3778,6 +3779,7 @@ function createClientSideRequest(history, location, signal, submission) {
     // won't be properly normalized to uppercase and results in a 405 error.
     // See: https://fetch.spec.whatwg.org/#concept-method
     init.method = formMethod.toUpperCase();
+    init.headers = submission.headers ? new Headers(submission.headers) : undefined;
     if (formEncType === "application/json") {
       init.headers = new Headers({
         "Content-Type": formEncType
