diff --git a/node_modules/react-router-dom/dist/dom.d.ts b/node_modules/react-router-dom/dist/dom.d.ts
index 0cfa784..10b5922 100644
--- a/node_modules/react-router-dom/dist/dom.d.ts
+++ b/node_modules/react-router-dom/dist/dom.d.ts
@@ -77,6 +77,11 @@ export interface SubmitOptions {
      * navigation when using the <ScrollRestoration> component
      */
     preventScrollReset?: boolean;
+    /**
+     * The headers to use when submitting the form. Warning: does not
+     * progressively enhance.
+     */
+    headers?: Headers | Record<string, string> | Array<Array<string>>;
 }
 export declare function getFormSubmissionInfo(target: SubmitTarget, basename: string): {
     action: string | null;
diff --git a/node_modules/react-router-dom/dist/index.js b/node_modules/react-router-dom/dist/index.js
index cf5f123..54b1d8a 100644
--- a/node_modules/react-router-dom/dist/index.js
+++ b/node_modules/react-router-dom/dist/index.js
@@ -757,6 +757,7 @@ function useSubmit() {
       body,
       formMethod: options.method || method,
       formEncType: options.encType || encType,
+      headers: options.headers,
       replace: options.replace,
       state: options.state,
       fromRouteId: currentRouteId
