diff --git a/node_modules/react-router-dom/dist/dom.d.ts b/node_modules/react-router-dom/dist/dom.d.ts
index cc979fc..1ec7365 100644
--- a/node_modules/react-router-dom/dist/dom.d.ts
+++ b/node_modules/react-router-dom/dist/dom.d.ts
@@ -85,6 +85,11 @@ export interface SubmitOptions {
      * navigation when using the <ScrollRestoration> component
      */
     preventScrollReset?: boolean;
+    /**
+     * The headers to use when submitting the form. Warning: does not
+     * progressively enhance.
+     */
+    headers?: Headers | Record<string, string> | Array<Array<string>>;
     /**
      * Enable flushSync for this navigation's state updates
      */
diff --git a/node_modules/react-router-dom/dist/index.js b/node_modules/react-router-dom/dist/index.js
index 1586c49..e1be510 100644
--- a/node_modules/react-router-dom/dist/index.js
+++ b/node_modules/react-router-dom/dist/index.js
@@ -1067,6 +1067,7 @@ function useSubmit() {
     if (options.navigate === false) {
       let key = options.fetcherKey || getUniqueFetcherId();
       router.fetch(key, currentRouteId, options.action || action, {
+        headers: options.headers,
         preventScrollReset: options.preventScrollReset,
         formData,
         body,
@@ -1076,6 +1077,7 @@ function useSubmit() {
       });
     } else {
       router.navigate(options.action || action, {
+        headers: options.headers,
         preventScrollReset: options.preventScrollReset,
         formData,
         body,
