diff --git a/node_modules/react-router-dom/dist/dom.d.ts b/node_modules/react-router-dom/dist/dom.d.ts
index c7b6421..95463f1 100644
--- a/node_modules/react-router-dom/dist/dom.d.ts
+++ b/node_modules/react-router-dom/dist/dom.d.ts
@@ -68,6 +68,11 @@ export interface SubmitOptions {
      * navigation when using the <ScrollRestoration> component
      */
     preventScrollReset?: boolean;
+    /**
+     * The headers to use when submitting the form. Warning: does not
+     * progressively enhance.
+     */
+    headers?: Headers | Record<string, string> | Array<Array<string>>;
 }
 export declare function getFormSubmissionInfo(target: HTMLFormElement | HTMLButtonElement | HTMLInputElement | FormData | URLSearchParams | {
     [name: string]: string;
diff --git a/node_modules/react-router-dom/dist/index.js b/node_modules/react-router-dom/dist/index.js
index ab16fb3..3cf0125 100644
--- a/node_modules/react-router-dom/dist/index.js
+++ b/node_modules/react-router-dom/dist/index.js
@@ -705,7 +705,8 @@ function useSubmitImpl(fetcherKey, routeId) {
       preventScrollReset: options.preventScrollReset,
       formData,
       formMethod: method,
-      formEncType: encType
+      formEncType: encType,
+      headers: options.headers
     };
 
     if (fetcherKey) {
