diff --git a/node_modules/react-router-dom/dist/dom.d.ts b/node_modules/react-router-dom/dist/dom.d.ts
index a26f274..defe243 100644
--- a/node_modules/react-router-dom/dist/dom.d.ts
+++ b/node_modules/react-router-dom/dist/dom.d.ts
@@ -70,6 +70,11 @@ interface SharedSubmitOptions {
      * navigation when using the <ScrollRestoration> component
      */
     preventScrollReset?: boolean;
+    /**
+     * The headers to use when submitting the form. Warning: does not
+     * progressively enhance.
+     */
+    headers?: Headers | Record<string, string> | Array<Array<string>>;
     /**
      * Enable flushSync for this submission's state updates
      */
diff --git a/node_modules/react-router-dom/dist/index.js b/node_modules/react-router-dom/dist/index.js
index c134e16..26e57ce 100644
--- a/node_modules/react-router-dom/dist/index.js
+++ b/node_modules/react-router-dom/dist/index.js
@@ -1076,6 +1076,7 @@ function useSubmit() {
     if (options.navigate === false) {
       let key = options.fetcherKey || getUniqueFetcherId();
       router.fetch(key, currentRouteId, options.action || action, {
+        headers: options.headers,
         preventScrollReset: options.preventScrollReset,
         formData,
         body,
@@ -1085,6 +1086,7 @@ function useSubmit() {
       });
     } else {
       router.navigate(options.action || action, {
+        headers: options.headers,
         preventScrollReset: options.preventScrollReset,
         formData,
         body,
